ts/main.ts

```diff
@@ -1 +1,36 @@
 /* global data */
+
+const form = document.getElementById('journal-entry-form') as HTMLFormElement;
+const photoUrlInput = document.getElementById('photo-url') as HTMLInputElement;
+const photoPreview = document.getElementById(
+  'photo-preview',
+) as HTMLImageElement;
+
+photoUrlInput.addEventListener('input', () => {
+  const url = photoUrlInput.value;
+  if (url) {
+    photoPreview.src = url;
+  } else {
+    photoPreview.src = 'images/placeholder-image-square.jpg';
+  }
+});
+
+form.addEventListener('submit', (event: Event) => {
+  event.preventDefault();
+
+  const titleInput = document.getElementById('title') as HTMLInputElement;
+  const notesInput = document.getElementById('notes') as HTMLTextAreaElement;
+
```

Comment:
If you need to query for elements, all of them should be global at the top of your page. This way, you query for them when the page loads. Currently you are running your query every time you submit the form which is a bit wasteful and will slow down your application.